### ISSUE TITLE
can't modify date in the database, put Unprocessable Entity

### ISSUE THREAD

**Issue by sayaa** [Automattic/mongoose#3581]
Hi everybody,

I use mongoose to handle my database and restangular to make crud in client size.
In my model, I have a parametre "comments" which is an array of object, and in each object I have a date
```javascript
var subSchema = new mongoose.Schema({
    date: {type: Date, default: Date.now},
    sender: String,
    content: String
}, {_id: false});

This subSchema is inserted in a schema after

var schema = new mongoose.Schema({   
    date: {type: Date, default: Date.now},
    title: String,
    comments: [subSchema]
}, {
    collection: 'test' 
});
```
something like this.

Everything is ok when I was using restangular for adding new documents.

But somewhere someone had updated some documents to add new comments using php. and I don't know why, the date he had send was wrong so now, some datas in the DB are corrupted,
I mean, I think mongoose or restangular don't recognise the format as a date, so when I make a read, it always give me today's date instead of the right one which where inserted.

now, when I look in my database, it gives me something like : 
```
{"0":{"content":"Test","sender":"saya","date":{"sec":"1446465106","usec":"299000"}}}
```
I tried to fix that by exporting all datas in a json file (cause when I use read of restangular it always returns today's date whiwh is wrong) and recreate the date :
something like : 

```javascript
...
var unix_timestamp = oPostFromJSON.comments[j].date.sec;
var json = JSON.stringify({$date: new Date(unix_timestamp * 1000)});
var date = JSON.parse(json);
var formatedDate = $filter('date')(date.$date, "yyyy/MM/dd HH:mm:ss Z");
 oPost.comments[i].date = formatedDate;

oPost.put().then(
       function(osuccessPut) {
       deffered.resolve(osuccessPut);
       },
       function(oErrorPut) {
       deffered.reject({error: oErrorPut, msg: 'put failed'});
       }
);
```
However, my put always fail

here is the error :

```
error: Object
    config: Object
    data: Array[1]
        0: Object
            kind: "cast"
            message: "Cast to date failed for value "[object Object]" at path "date""
            name: "ValidatorError"
            path: "date"
            properties: Object
            stack: "Cast to date failed for value "[object Object]" at path "date"↵ at init (/www/api/node_modules/mongoose/lib/document.js:312:37)↵ at EmbeddedDocument.Document.init (/www/api/node_modules/mongoose/lib/document.js:268:3)↵ at DocumentArray.cast (/www/api/node_modules/mongoose/lib/schema/documentarray.js:185:27)↵ at init (/www/api/node_modules/mongoose/lib/document.js:310:29)↵ at model.Document.init (/www/api/node_modules/mongoose/lib/document.js:268:3)↵ at completeOne (//www/api/node_modules/mongoose/lib/query.js:1453:10)↵ at Object.<anonymous> (/www/api/node_modules/mongoose/lib/query.js:1195:13)↵ at Object._onImmediate (/www/api/node_modules/mongoose/node_modules/mquery/lib/utils.js:137:16)↵ at processImmediate [as _immediateCallback] (timers.js:345:15)"
            value: Object
                sec: 1446465106
                usec: 299000
                __proto__: Object
               __proto__: Object
        length: 1
        __proto__: Array[0]
    headers: (name)
    status: 422
    statusText: "Unprocessable Entity"
```

the value object 
(value: Object
sec: 1446465106
usec: 299000)
 is the value already in the document and which I would like to modify. but it never works.
one solution I found was to remove the corrupted document and insert the correct one (the one I modified from the json file), 
but I can't use that solution because the _id was modified even if it is specified in the new object I send to the database and I need the _id to be unchanged. :(
I've also tried to modify the date by setting its value to null in the way to modify it with the right value after that but it doesn't work at all, I alwas have the same cast error and: "PUT Unprocessable Entity"
I don't know what to do, please help.

**Comment by vkarpov15** [comment]
Which version of mongoose are you using? Mongoose 3.8 makes it impossible to save the doc once you've encountered a cast error. This has since been fixed.

Easiest solution would be to just run a script directly against the database to update all the docs - do you have access to do that?

**Comment by sayaa** [comment]
yes, I have. I'll try that solution then. Will tell you if it works

**Comment by sayaa** [comment]
it works, I've run my script directly against my database for my update. thank's for the idea.

but I have one last question, let's imagine that I hadn't the access to make my updates directly against the database, am I obliged to change mongoose's version? if so, which version should I use?

**Comment by vkarpov15** [comment]
4.2.7 is the latest, I'd recommend using that where possible. The cast error behavior I described is difficult to fix in mongoose 3.8.x without backwards breaking changes, which is why it's only fixed in >= 4.0.0.
